contracts/sfc/SFCLib.sol

```diff
@@ -404,8 +408,14 @@ contract SFCLib is SFCBase {
         _burnFTM(amount);
     }
 
+    function burnNativeTokens() external payable {
+        require(msg.value > 0, "No amount sent");
+        _burnFTM(msg.value);
+    }
+
     function _burnFTM(uint256 amount) internal {
-        if (amount != 0) {
+        if (amount != 0 && totalSupply >= amount) {
```

Comment:
What if `totalSupply < amount`? The method they silently fail, but claims everthing is ok.